test/dns-test.js

```diff
@@ -0,0 +1,396 @@
+
+    for (let i = 0; i < 15; i++)
+      await nclient.execute('generatetoaddress', [2, coinbase]);
+  });
+
+  after(async () => {
+    await nclient.close();
+    await wclient.close();
+    await node.close();
+    await nstub.close();
+    await rstub.close();
+  });
+
+  describe('Recursive Resolver', () => {
+    it('should return an A record', async () => {
+      const name = 'google.com.';
+
+      const res = await rstub.lookup(name, types.A);
+
+      const [question] = res.question;
+      assert(typeof question === 'object');
+      assert.equal(question.name, name);
+
+      const [answer] = res.answer;
+      assert(typeof answer === 'object');
+      assert.equal(answer.name, name);
+    });
+  });
+
+  describe('Authoritative Resolver', () => {
+    it('should return no answer for an ICANN rooted name', async () => {
+      const name = 'com.';
+
+      const res = await nstub.lookup(name, types.A);
+
+      assert.equal(res.question[0].name, name);
+      assert.equal(res.answer.length, 0);
+
+      // Handshake zone is not authoritative for ICANN based name
+      assert.equal(res.aa, false);
+    });
+
+    it('should return no answer for HNS name pre-update', async () => {
+      const name = 'foo.';
+
+      const res = await nstub.lookup(name, types.A);
+      assert.equal(res.answer.length, 0);
+    });
+
+    // record to save in the authenticated tree
+    for (const [hstype, item] of Object.entries(json)) {
+      const type = types[item.type];
+      const resource = Resource.fromJSON(item.resource);
+
+      // args: size, height, network
+      const name = rules.grindName(5, 1, network);
+
+      it(`should update ${name}`, async () => {
+        await wallet.client.post(`/wallet/${wallet.id}/open`, {
+          name: name
+        });
+
+        await mineBlocks(treeInterval + 1, coinbase);
+
+        await wallet.client.post(`/wallet/${wallet.id}/bid`, {
+          name: name,
+          bid: 1000,
+          lockup: 2000
+        });
+
+        await mineBlocks(biddingPeriod + 1, coinbase);
+
+        await wallet.client.post(`/wallet/${wallet.id}/reveal`, {
+          name: name
+        });
+
+        await mineBlocks(revealPeriod + 1, coinbase);
+
+        await wallet.client.post(`/wallet/${wallet.id}/update`, {
+          name: name,
+          data: resource.toJSON()
+        });
+
+        // mine a block
+        await mineBlocks(1, coinbase);
+
+        const rinfo = await nclient.execute('getnameresource', [name]);
+
+        const json = resource.getJSON(name);
+        assert.deepEqual(rinfo, json);
+
+        const info = await nclient.execute('getnameinfo', [name]);
+        const data = Buffer.from(info.info.data, 'hex');
+
+        const returned = Resource.decode(data);
+        assert.deepEqual(returned.getJSON(name), json);
+
+        // this commits the tree state to disk
+        // updates served by the dns server
+        await mineBlocks(treeInterval, coinbase);
```

Comment:
There's a few places around the testing suite where a complete auction process must be run, I think there's also one that uses RPC calls like this. Anyway, that code is _covered_, and is a bit of a waste here, especially since every one of these tests takes ~500ms on my machine. 

I think we should find a simple-unit-test way to just inject a single record into the tree without any of the blockchain BS. This entire test could be 90% lighter. Maybe it's possible to even drop the `FullNode` entirely and just run the StubResolvers, RootServer, and NameServer, with a hacked-in urkel tree.

Reply:
Looks like `FullNode` instantiates a `RootResolver` with the option:
`lookup: key => this.chain.db.tree.get(key)`
I wonder if that can just be stubbed with a simple map. All the updates in this test could be simple insertions!

Reply:
Going to replace the whole auction process with just writing to the tree directly. This will speed up the tests, remove the need for a bunch of code and better decouple the tests. Good feedback